From: Simon McVittie <smcv@debian.org>
Date: Sat, 4 Mar 2017 15:15:50 +0000
Subject: glib-mkenums: Sort input files for more deterministic output

This should be helpful for reproducible builds
<https://reproducible-builds.org/>.

Perl's sorting is not locale-sensitive unless the lexical scope has
'use locale', which this one does not, so we do not need to force
locale-agnostic sorting.

Signed-off-by: Simon McVittie <smcv@debian.org>
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=769983
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=809152
Forwarded: yes
---
 gobject/glib-mkenums.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gobject/glib-mkenums.in b/gobject/glib-mkenums.in
index f7da97506..219a16615 100755
--- a/gobject/glib-mkenums.in
+++ b/gobject/glib-mkenums.in
@@ -290,6 +290,8 @@ if (length($fhead)) {
     print "$prod\n";
 }
 
+@ARGV = sort @ARGV;
+
 while (<>) {
     if (eof) {
 	close (ARGV);		# reset line numbering
